### PR TITLE
feat: use pip environmental variables for repository urls

### LINKED ISSUE #2181: How do I add support for different pypi providers?
#### This is a:

- [ ] Bug report (non-security related)
- [ ] Feature request
- [x] I'm not sure which of those it is

#### I'm using:

- [x] The Renovate GitHub App
- [ ] Self-hosted GitHub
- [ ] Self-hosted GitLab
- [ ] Self-hosted VSTS

#### Please describe the issue:

I see that [public pypi](https://github.com/renovatebot/renovate/blob/master/lib/datasource/pypi.js#L12) is the sensible default. I'd like to make it so that either `PIP_INDEX_URL` from the environment or `--pip-index-url` from the command line would override the host, allowing for private/other hosts.

Where should I get started in writing the test for that?

### PR DESCRIPTION
**Pull request by khornberg**
Adds support for PIP_INDEX_URL from environmental variables for self
hosted version. This is used instead of any urls set in the config.

Closes #2181

### CODE REVIEW

File: test/datasource/pypi.spec.js [hunk 1]
```diff
@@ -39,6 +39,14 @@ describe('datasource/pypi', () => {
       await datasource.getDependency('pkg:pypi/azure-cli-monitor', config);
       expect(got.mock.calls).toMatchSnapshot();
     });
+    it('supports custom datasource url from environmental variable', async () => {
+      got.mockReturnValueOnce({
+        body: JSON.parse(res1),
+      });
+      process.env.PIP_INDEX_URL = 'https://my.pypi.python/pypi/'
```
Comment (hbetts) [inline test/datasource/pypi.spec.js]: Does this leak to other tests?

Comment (rarkins) [reply test/datasource/pypi.spec.js]: I suspect process.env is per-file however it would be good to save and replace here just to be safe.

Comment (khornberg) [reply test/datasource/pypi.spec.js]: By save and replace do you mean to save the value of PIP_INDEX_URL to a variable before setting it and then set the value back after the test; or do you mean unset the variable altogether after the test?

Comment (rarkins) [reply test/datasource/pypi.spec.js]: Probably either is OK, but the first one is most foolproof

**Comment by khornberg** [comment]
I don't understand the CI failure. Is there something I can do to fix it?

**Comment by rarkins** [comment]
@khornberg I think it's just prettier formatting. You can run `yarn prettier` locally to fix.

**Comment by khornberg** [comment]
Awesome thanks

**Pull request merged**